crates/astria-composer/src/executor/mod.rs

```diff
@@ -276,6 +290,37 @@ impl Executor {
                     }
                 }
             }
+        };
+
+        match reason {
+            Ok(reason) => {
+                info!(%reason, "draining remaining bundles");
+                // when shutting down, drain all the remaining bundles and submit to the sequencer
+                // to avoid any bundle loss.
+                let mut bundles_to_drain: Vec<SizedBundle> = vec![];
+                // drain all the bundles from the factory
+                loop {
+                    let bundle = bundle_factory.pop_now();
+                    if bundle.is_empty() {
+                        break;
+                    }
+                    bundles_to_drain.push(bundle);
+                }
+                let num_bundles_to_drain = bundles_to_drain.len();
+
+                // we need to submit the bundles sequentially since that is the guarantee
+                // the bundle factory gives to the end users.
+                for bundle in bundles_to_drain {
+                    let res = self.submit_bundle(nonce, bundle).await?;
+                    nonce = res;
```

Comment:
This entire process should be guarded by a timeout. If your 25 (or 30) seconds are reached, you don't want to exit here uncontrolled. It's better to try and submit and - if it failed - report the txes that could not be submitted.

Also, please no `?` without context.

Reply:
will remove the `?`. I ll add a timeout here. I initially thought we could just have the timeout in the Composer, but i think it makes sense for reporting the bundles which we lost.